ng2-components/ng2-activiti-tasklist/src/components/adf-task-attachment-list.component.ts

```diff
@@ -90,31 +97,68 @@ export class TaskAttachmentListComponent implements OnChanges {
     }
 
     onShowRowActionsMenu(event: any) {
-        let myAction = {
-            title: 'Delete',
-            name: 'delete'
+        let viewAction = {
+            title: 'View',
+            name: 'view'
         };
+
+        let removeAction = {
+            title: 'Remove',
+            name: 'remove'
+        };
+
+        let downloadAction = {
+            title: 'Download',
+            name: 'download'
+        };
+
         event.value.actions = [
-            myAction
+            viewAction,
+            removeAction,
+            downloadAction
         ];
     }
 
     onExecuteRowAction(event: any) {
         let args = event.value;
         let action = args.action;
-        if (action.name === 'delete') {
+        if (action.name === 'view') {
+            this.emitDocumentContent(args.row.obj);
+        } else if (action.name === 'remove') {
             this.deleteAttachmentById(args.row.obj.id);
+        } else if (action.name === 'download') {
+            this.downloadContent(args.row.obj);
         }
     }
 
     openContent(event: any): void {
         let content = event.value.obj;
+        this.emitDocumentContent(content);
+    }
+
+    emitDocumentContent(content: any) {
         this.activitiContentService.getFileRawContent(content.id).subscribe(
             (blob: Blob) => {
                 content.contentBlob = blob;
                 this.attachmentClick.emit(content);
+            },
+            (err) => {
+                this.error.emit(err);
             }
         );
     }
 
+    downloadContent(content: any): void {
+        this.activitiContentService.getFileRawContent(content.id).subscribe(
+            (blob: Blob) => this.contentService.downloadBlob(blob, content.name),
+            (err) => {
+                this.error.emit(err);
+            }
+        );
+    }
+
+    attachDocumentToTask() {
```

Comment:
It is wrong for me, we should emit the attachmentCreate when the upload is done not just when the button is clicked.